interactions/api/models/guild.py

```diff
@@ -236,6 +239,144 @@ def __init__(self, **kwargs):
             else None
         )
 
+    async def ban(
```

Comment:
This is awesome, but if the ultimate goal of the helper methods here is to help add recursion to what d.py had in a similar fashion, this would make better sense to include in the `Member` object as well. having `ctx.guild.ban()` and `ctx.guild.members[member].ban()` will both be very convenient.

Reply:
`ctx,guild.members`  is only sent in the `GUILD_CREATE` event https://discord.com/developers/docs/resources/guild#guild-object , so I don't see the sense to create a method for it. Also I wanted to implement it directly into the `Member` object, but since it has no `guild_id` attribute I couldn't

Reply:
That doesn't mean it shouldn't be added in only because 1 gateway event has it. During the startup of the bot, any `GUILD_CREATE` events dispatched by the gateway before the client is internally ready, they're cached into the global `Cache` as a K:V in `self_guilds`. This also generally goes against using this paradigm logic:

```py
for member in ctx.guild.members:
    await member.ban()
```